astropy/coordinates/builtin_frames/utils.py

```diff
@@ -34,29 +35,18 @@
 _DEFAULT_PM = (0.035, 0.29)*u.arcsec
 
 
-def cartrepr_from_matmul(pmat, coo, transpose=False):
-    """
-    Note that pmat should be an ndarray, *not* a matrix.
+def matrix_product(*args):
+    """Matrix multiply all arguments together.
+
+    Arguments should have dimension 2 or larger. Larger dimensional objects
+    are interpreted as stacks of matrices residing in the last two dimensions.
```

Comment:
I think this docstring needs a bit more about *why* this function exists (which if I'm understanding right is being there mainly to ensure a consistent convention for dimensionality and so on).